### PR TITLE
vimc-3404: simpler/better test mode

### PR DESCRIPTION
**Pull request by richfitz**
This augments the current test mode (`orderly::orderly_test_start`).  See the included docs for explanation as this needs to be clear enough to standalone...

Things for later tickets

* There is some interaction with `orderly::orderly_run_info` in that we can't easily run this while interacting with the new `orderly_develop_*` functions
* I am resisting the urge to add other features that might build off this (such as creating templates for resource, dependencies and artefacts sections based on the unknown files found in the report)
* There is still a bootstrapping problem in that we can't even read the recipe for an `orderly.yml` that does not have a script or an artefacts section, but we should be able to really
* Printing the status in a way that looks better
* Interaction with git so that we can require that a directory gitignores all files that would be copied into it *before* they are copied

### CODE REVIEW

File: R/develop.R [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+##'
+##' Some degree of care should be used while using these functions.
+##'
+##' Because \code{orderly_develop_start} copies files into your source
+##' tree you should be careful to add these files to your
+##' \code{.gitignore} files so that they are not included if using
+##' git.  Rerunning \code{orderly_develop_start} will copy a fresh
+##' copy dependencies into your tree, overwriting files that are there
+##' without warning.
+##'
```
[the remark this answers is not in the document]
Comment (richfitz) [reply R/develop.R]: missing `of`

**Comment by richfitz** [comment]
I'll do the rebase after the next review, or we'll be doing it over and over...

File: R/develop.R [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                                     FALSE, FALSE))
+  status <- data_frame(
+    filename = unlist(internal, FALSE, FALSE),
+    type = rep(names(internal), lengths(internal)))
+  extra <- setdiff(dir(path, recursive = TRUE), status$filename)
+  if (length(extra) > 0L) {
+    status <- rbind(status, data_frame(filename = extra, type = "unknown"))
+  }
+  status$present <- file.exists(file.path(path, status$filename))
+  ## TODO: better name?
```
Comment (r-ash) [inline R/develop.R]: Old TODO still around? `derived` makes sense to me as a name

Comment (richfitz) [reply R/develop.R]: ok, I wasn't sold but haven't come up with a better one - I'll remove the notice

File: inst/init/orderly.yml [hunk 3]
```diff
@@ -20,12 +20,7 @@
 #       another: from_file.sql
 #
 # If you are writing a report that uses only artefacts from other
-# reports, the data field can be empty, as:
-#
-#     data: ~
-#
-# but it must still exist.
-data: ~
+# reports, you can omit this field.
```
[the remark this answers is not in the document]
Comment (richfitz) [reply inst/init/orderly.yml]: thanks - nice catch

**Pull request merged**